From e7148d9e0f5410e4572adf540764d82fff6aa2b6 Mon Sep 17 00:00:00 2001
From: Daniel Kahn Gillmor<dkg@fifthhorseman.net>
Date: Wed, 07  Jan  2009  00:55:15  +0000
Subject: backport fix to deprecate MD5 hashes during X.509 certificate validation

** gnutls: deprecate X.509 validation chains using MD5 and MD2 signatures.
This is a bugfix -- the previous attempt to do this from internal x509
certificate verification procedures did not return the correct value
for certificates using a weak hash.

http://thread.gmane.org/gmane.comp.encryption.gpg.gnutls.devel/3332

diff --git a/lib/x509/verify.c b/lib/x509/verify.c
index be01b5f..a644b13 100644
--- a/lib/x509/verify.c
+++ b/lib/x509/verify.c
@@ -320,6 +320,7 @@ _gnutls_verify_certificate2 (gnutls_x509_crt_t cert,
 	{
 	  if (output)
 	    *output |= GNUTLS_CERT_INSECURE_ALGORITHM | GNUTLS_CERT_INVALID;
+	  ret = 0;
 	}
     }
 
@@ -1036,6 +1037,7 @@ _gnutls_verify_crl2 (gnutls_x509_crl_t crl,
       {
 	if (output)
 	  *output |= GNUTLS_CERT_INSECURE_ALGORITHM | GNUTLS_CERT_INVALID;
+	ret = 0;
       }
   }
 
--
cgit v0.8.1
